service/src/main/java/bio/terra/externalcreds/dataAccess/SshKeyPairDAO.java

```diff
@@ -60,11 +60,15 @@ public SshKeyPairInternal upsertSshKeyPair(SshKeyPairInternal sshKeyPairInternal
             + " external_user_email = excluded.external_user_email"
             + " RETURNING id";
 
+    var sshPrivateKey = sshKeyPairInternal.getPrivateKey();
+    if (externalCredsConfig.getKmsConfiguration().isPresent()) {
+      sshPrivateKey = kmsEncryptDecryptHelper.encryptSymmetric(sshPrivateKey);
+    }
```

Comment:
2 comments:
1) style - I would rather have this be 
```suggestion
    var sshPrivateKey;
    if (externalCredsConfig.getKmsConfiguration().isPresent()) {
      sshPrivateKey = kmsEncryptDecryptHelper.encryptSymmetric(sshKeyPairInternal.getPrivateKey());
    } else {
      sshPrivateKey = sshKeyPairInternal.getPrivateKey();
    }
```

2) This pattern means that all callers of `kmsEncryptDecryptHelper` need to do this check. Can you push it into the `kmsEncryptDecryptHelper` instead? In other words, if KMS is not configured `kmsEncryptDecryptHelper` just returns what is passed in and does not throw an exception. Maybe log a message at startup that encryption is enabled or disable.